### ISSUE TITLE
Recursive issue

### ISSUE THREAD

**Issue by goran-mahovlic** [Ache17/fluorescence_spectrometer#1]
Hi, I am adding this sensor to lora generic node, and have notticed this

https://github.com/Ache17/fluorescence_spectrometer/blob/2089409c700b6a3a50c98ad53aaa42decc147079/src/main.c#L252

It might be wrong as I doubt you wanted to go recursive here.

Did you get good readings from your sensor? I do get something, but think some values are wrong for some reason.

**Comment by mlewandowski0** [comment]
Yeah you are right about that, code I have pushed was just for other teammates to look into it ( we did correct it in the lab ).
I will push the correct code soon if you want ( never thought it will give any interest to anyone really ).
The project is work in progress, we experiment quite a lot so the repository remained dead a bit ( too much weird, sometimes not related code, we will have a release soon though ). 

We did get a reasonable measurements ( we have had though some weird problems with the sensor, might just be a particular board) though the code is not released yet, sorry.

**Comment by goran-mahovlic** [comment]
Oh, do not be sorry, as your code helped me a lot, I did not need to port everything by myself
So then you for putting it on even if it is experimental.

https://github.com/goran-mahovlic/generic-node-se/blob/develop/Software/lib/AS7341/AS7341.c

Those are my current readings, I get strange - I would say inverted readings on ADC 2 and 3

![read](https://user-images.githubusercontent.com/4248736/110508315-2fb0d300-8101-11eb-89e2-540ddb766d23.png)
